src/main/flight/pid.c

```diff
@@ -86,13 +90,30 @@ void pidResetErrorAngle(void)
 
 void pidResetErrorGyro(void)
 {
-    errorGyroI[ROLL] = 0;
-    errorGyroI[PITCH] = 0;
-    errorGyroI[YAW] = 0;
+    int axis;
+    for (axis = 0; axis < 3; axis++) {
+        errorGyroI[axis] = 0;
+        errorGyroIf[axis] = 0.0f;
+    }
+}
+
+float scaleItermToRcInput(int axis) {
+    float rcCommandReflection = (float)rcCommand[axis] / 500.0f;
+    static float iTermScaler[3] = {1.0f, 1.0f, 1.0f};
+
+    if (ABS(rcCommandReflection) > 0.7f && (!flightModeFlags)) {   /* scaling should not happen in level modes */
```

Comment:
(and add it close to flightModeFlags_e definition, so that it can be updated when new mode is added)